geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/ProtobufOpsProcessor.java

```diff
@@ -52,7 +52,7 @@ public ProtobufOpsProcessor(SerializationService serializationService,
     ClientProtocol.Response.Builder builder;
     Result result;
     try {
-      if (context.getAuthorizer().authorize(operationContext.getAccessPermissionRequired())) {
+      if (context.authorize(operationContext.getAccessPermissionRequired())) {
```

Comment:
To keep `MessageExecutionContext` purely a data container, use something like `if (context.getAuthorizer().authorize(context.getSubject(), operationContext.getAccessPermissionRequired())) {` instead.

Reply:
Ugh, I hate that the authorization internal state is leaking out like this, but I agree that trying to put the authorization logic on the MessageExecutionContext isn't the right solution.  I think a better solution would have been to have captured this state in the authenticator, but not sure it's worth the discussion and refactoring to get that change made at this point, so I'll go with the code you proposed here.